### PR TITLE
GPT review on comment

### PR DESCRIPTION
**Pull request by ccubu**

### CODE REVIEW

File: .github/workflows/chatgpt.yml [hunk 1]
```diff
@@ -17,30 +17,18 @@ jobs:
...
-          # Fetch the latest commit message associated with this PR
-          LATEST_COMMIT_MESSAGE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
-            https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/commits | \
-            jq -r '.[-1].commit.message')
-          echo "Latest Commit Message: $LATEST_COMMIT_MESSAGE"
-          # Fetch the PR creator
-          PRCREATOR=$(curl -s -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.pull_request.number }} | jq -r .user.login)
+          # Check if the comment contains [autoreview]
+          COMMENT_BODY="${{ github.event.comment.body }}"
+          echo "is_autoreview=$(if [[ "$COMMENT_BODY" == *"[autoreview]"* ]]; then echo 'true'; else echo 'false'; fi)" >> $GITHUB_ENV
```
Comment (dastanfv) [inline .github/workflows/chatgpt.yml:22]:
```suggestion
          echo "is_gpt_review=$(if [[ "$COMMENT_BODY" == *"[gpt review]"* ]]; then echo 'true'; else echo 'false'; fi)" >> $GITHUB_ENV
```

**Comment by dastanfv** [comment]
[autoreview]

**Comment by ccubu** [comment]
> [autoreview]

As far as I understood reading the docs and other users, it won't trigger until it's merged and therefore the workflow is on the default branch

**Comment by dastanfv** [comment]
> > [autoreview]
> 
> As far as I understood reading the docs and other users, it won't trigger until it's merged and therefore the workflow is on the default branch

yeh, just wanted to try

File: .github/workflows/chatgpt.yml [hunk 2]
```diff
@@ -17,30 +17,18 @@ jobs:
...
+          # Check if the commenter is in the specified group
+          COMMENTER="${{ github.event.comment.user.login }}"
+          IS_IN_GROUP=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+            https://api.github.com/orgs/freeverseio/teams/ziggy/members/$COMMENTER)
+          echo "is_in_group=$(if [[ "$IS_IN_GROUP" != "null" ]]; then echo 'true'; else echo 'false'; fi)" >> $GITHUB_ENV
 
-      - name: Run if Draft, Correct User, and AutoReview
-        if: env.is_draft == 'true' && env.is_in_group == 'true' && env.is_autoreview == 'true'
+      - name: Run if Correct User, and AutoReview
+        if: env.is_in_group == 'true' && env.is_autoreview == 'true'
```
Comment (dastanfv) [inline .github/workflows/chatgpt.yml:31]:
```suggestion
        if: env.is_in_group == 'true' && env.is_gpt_review == 'true'
```

File: .github/workflows/chatgpt.yml [hunk 3]
```diff
@@ -17,30 +17,18 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          # Fetch the latest commit message associated with this PR
-          LATEST_COMMIT_MESSAGE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
-            https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/commits | \
-            jq -r '.[-1].commit.message')
-          echo "Latest Commit Message: $LATEST_COMMIT_MESSAGE"
-          # Fetch the PR creator
-          PRCREATOR=$(curl -s -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.pull_request.number }} | jq -r .user.login)
+          # Check if the comment contains [autoreview]
```
Comment (dastanfv) [inline .github/workflows/chatgpt.yml:20]:
```suggestion
          # Check if the comment contains "gpt review"
```

**Review by dastanfv — COMMENTED** [review]
final comments

File: .github/workflows/chatgpt.yml (hunk 2 above)
Comment (ccubu) [reply .github/workflows/chatgpt.yml]: done

File: .github/workflows/chatgpt.yml (hunk 3 above)
Comment (ccubu) [reply .github/workflows/chatgpt.yml]: done

**Review by dastanfv — APPROVED** [review]

**Comment by dastanfv** [comment]
gpt review please

**Pull request merged**